Add throttling for statement log (#24642)
MaterializeInc/materialize: doc/user/content/sql/system-catalog/mz_internal.md

| `redacted_sql`            | [`text`]                     | The SQL text of the statement, in a normalized form, with all string and numeric literals hidden.                                                                                                                                                                             |
| `prepared_at`             | [`timestamp with time zone`] | The time at which the statement was prepared.                                                                                                                                                                                                                                 |
| `statement_type`          | [`text`]                     | The _type_ of the statement, e.g. `select` for a `SELECT` query, or `NULL` if the statement was empty.                                                                                                                                                                        |

> | `throttled_count`         | [`uint8`]                    | The number of statements that were dropped due to throttling before this one was seen. If you have a very high volume of queries and need to log them without throttling, please contact support.                                                                             |

Reviewer: ```suggestion
| `throttled_count`         | [`uint8`]                    | The number of statements that were dropped due to throttling before the current one was seen. If you have a very high volume of queries and need to log them without throttling, [contact our team](https://materialize.com/docs/support/).                                                                             |
```

Reviewer: Doesn't feel like the ideal place to have the callout, but we can course-correct once we make the statement log prominent elsewhere in the docs.

| `redacted_sql`            | [`text`]                     | The SQL text of the statement, in a normalized form, with all string and numeric literals hidden.                                                                                                                                                                             |
| `prepared_at`             | [`timestamp with time zone`] | The time at which the statement was prepared.                                                                                                                                                                                                                                 |
| `statement_type`          | [`text`]                     | The _type_ of the statement, e.g. `select` for a `SELECT` query, or `NULL` if the statement was empty.                                                                                                                                                                        |

> | `n_throttled`             | [`uint8`]                    | The number of statements that were dropped due to throttling before this one was seen. If you have a very high volume of queries and need to log them without throttling, please contact support.                                                                             |

Reviewer 2: `uint8`s are not understandable by many drivers because it's not a native postgres oid type. I encourage the use of int8 here instead.

Author: We're already using `uint8` for various other things in this view (at least `execution_timestamp` and `transaction_id`). I didn't want to introduce an inconsistency for just this one field -- maybe we should consider switching those all over to `int8` at the same time.

Btw, do you know anything about why we support `uint8` at all, given that it's not commonly supported?
